src/local.rs

```diff
@@ -113,14 +114,14 @@ impl<W: Write> WriteClose for NopDeferredDelete<W> {}
 impl WriteClose for NamedTempFile {}
 
 /// Generate a script that runs kubecfg show and kubectl apply and runs it.
-pub fn apply(
+pub async fn apply(
     app_instance: &str,
     dry_run: &Option<DryRun>,
     package_image: &Option<String>,
     impersonate_user: &Option<String>,
     pre_diff: bool,
     is_local: bool,
-) -> Result<()> {
+) -> Box<dyn Future<Output = ()>, Result<()>> {
```

Comment:
I'm very surprised you need this.

Reply:
This was me following the compiler for a WIP commit, but it does run into another error because this is unstable.

```
error[E0658]: use of unstable library feature 'allocator_api'
   --> src/local.rs:124:35
    |
124 | ) -> Box<dyn Future<Output = ()>, Result<()>> {
    |                                   ^^^^^^^^^^
```

Reply:
the right thing would be
```
Box<dyn Future<Output = Result<()>>> {
```

but that's not necessary because you marked the fn as `async`, and that's what happens under the hood

Reply:
I created a [PR](https://github.com/kubecfg/kubit/pull/177) against this branch that fixes this. The issue was related to recursive async calls [E0733](https://doc.rust-lang.org/error_codes/E0733.html). I just refactored the code (probably could be done better) to prevent the need for recursion.

Reply:
I think I understand this.

@jeffreyssmith2nd If you don't mind, can I try digest what you did in that PR and incorporate it into this one on Monday?

Although if this feature is needed urgently then it's no problem to close this one out in favour of yours so we can push it through.

Reply:
If the PR is against this branch it can be merged here

Reply:
You're right, I missed that completely when skimming on mobile.

Massive thanks @jeffreyssmith2nd :rocket: